js/plugins/anthropic/src/models.ts

```diff
@@ -0,0 +1,371 @@
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type Anthropic from '@anthropic-ai/sdk';
+import type {
+  GenerateRequest,
+  GenerateResponseData,
+  ModelReference,
+  StreamingCallback,
+} from 'genkit';
+import { z } from 'genkit';
+import type { GenerateResponseChunkData, ModelAction } from 'genkit/model';
+import { modelRef } from 'genkit/model';
+import { model } from 'genkit/plugin';
+
+import type { ModelInfo } from 'genkit/model';
+import { BetaRunner, Runner } from './runner/index.js';
+import {
+  AnthropicBaseConfigSchema,
+  AnthropicBaseConfigSchemaType,
+  AnthropicConfigSchema,
+  AnthropicThinkingConfigSchema,
+  resolveBetaEnabled,
+  type ClaudeModelParams,
+  type ClaudeRunnerParams,
+} from './types.js';
+
+// This contains all the Anthropic config schema types
+type ConfigSchemaType =
+  | AnthropicBaseConfigSchemaType
+  | AnthropicThinkingConfigSchemaType;
+
+/**
+ * Creates a model reference for a Claude model.
+ * Computes the default version from info.versions array and sets it on the modelRef.
+ */
+function commonRef(
+  name: string,
+  info?: ModelInfo,
+  configSchema: ConfigSchemaType = AnthropicConfigSchema
+): ModelReference<ConfigSchemaType> {
+  // Compute default version from info.versions array
+  let defaultVersion: string | undefined;
+  if (info?.versions && info.versions.length > 0) {
+    // Prefer version with '-latest' suffix
+    const latestVersion = info.versions.find((v) => v.endsWith('-latest'));
+    if (latestVersion) {
+      defaultVersion = latestVersion;
+    } else if (info.versions.includes(name)) {
+      // If base name exists in versions array, use it directly
+      defaultVersion = name;
+    } else {
+      // Otherwise use first version
+      defaultVersion = info.versions[0];
+    }
+  }
+
+  return modelRef({
+    name: `anthropic/${name}`,
+    configSchema,
+    version: defaultVersion,
+    info: info ?? {
+      supports: {
+        multiturn: true,
+        tools: true,
+        media: true,
+        systemRole: true,
+        output: ['text'],
+      },
+    },
+  });
+}
+
+export const KNOWN_CLAUDE_MODELS: Record<
+  string,
+  ModelReference<
+    AnthropicBaseConfigSchemaType | AnthropicThinkingConfigSchemaType
+  >
+> = {
+  'claude-3-haiku': commonRef(
+    'claude-3-haiku',
+    {
+      versions: ['claude-3-haiku-20240307'],
+      label: 'Anthropic - Claude 3 Haiku',
+      supports: {
```

Comment:
The reason to have commonRef is so you don't have to pass all the same information over and over. Most models have the same info, that's why commonRef has a default. So you don't have to pass the default into the function. (Have a look at vertexai/src/modelgarden/anthropic.ts.)